test/parse.js

```diff
@@ -310,3 +310,14 @@ test('query strings having comma encoded and format option as `comma`', t => {
 		]
 	});
 });
+
+test('value shoud not be decoded twice with arrayFormat set as `separator`', t => {
+	t.deepEqual(queryString.parse('foo=2020-01-01T00:00:00%2B03:00', {arrayFormat: 'separator'}), {
+		foo: '2020-01-01T00:00:00+03:00'
+	});
+});
+
+// See https://github.com/sindresorhus/query-string/pull/243#issuecomment-591179715
+test.failing('value separated by encoded comma will not be parsed as array with arrayFormat set as `comma`', t => {
+	t.fail();
```

Comment:
This should actually be a valid test implementation, which would pass when the issue is fixed.